app/models/user.rb

```diff
@@ -0,0 +1,50 @@
+class User < ApplicationRecord
+  before_create :generate_confirmation_token
+  after_create_commit :send_verification_mail
+
+  enum role: {
+    'admin' => 0,
+    'user' => 1
+  }
+
+  has_secure_password
+
+  validates :name, :email, presence: true
+  validates :password, :password_confirmation, presence: true, if: :setting_password?
+  validates :email, uniqueness: true
+  validates :email, format: { with: QuoraClone::RegexConstants::EMAIL_REGEX }, allow_blank: true
+
+  def update_password_reset_token
+    if update(password_reset_token: TokenHandler.generate_token, reset_password_sent_at: Time.current)
```

Comment:
extract update and send mail in methods and merge this branch

Reply:
discuss